src/Compilers/Core/Portable/AssemblyUtilities.cs

```diff
@@ -105,7 +105,7 @@ public static ImmutableArray<string> FindSatelliteAssemblies(string filePath)
 
             var builder = ImmutableArray.CreateBuilder<string>();
 
-            string? directory = Path.GetDirectoryName(filePath);
+            string directory = Path.GetDirectoryName(filePath)!;
```

Comment:
Our guideline is to only use `!` if there is logic in the method that already checks in some way.
This should be an assertion instead.

Reply:
Well, for pathes thats are combined, getting directory should be safe. Assertion is OK.